lib/logidze/meta.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'logidze/pg_helpers'
+
+module Logidze # :nodoc:
+  # Provide methods to attach meta information
+  module Meta
+    def with_meta(meta, &block)
+      MetaTransaction.wrap_with(meta, &block)
+    end
+
+    def with_responsible(responsible_id, &block)
+      return yield if responsible_id.nil?
+      meta = { Logidze::History::Version::RESPONSIBLE => responsible_id }
+      with_meta(meta, &block)
+    end
+
+    class MetaTransaction # :nodoc:
+      include Logidze::PgHelpers
+
+      def self.wrap_with(meta, &block)
+        new(meta, &block).perform
+      end
+
+      attr_reader :meta, :block
+
+      def initialize(meta, &block)
+        @meta = meta
+        @block = block
+      end
+
+      def perform
+        return block.call if meta.nil?
+
+        ActiveRecord::Base.transaction do
+          set_meta_param(prev_meta.merge(meta))
+          result = block.call if block
+          reset_meta_param
+          result
+        end
+      end
+
+      private
+
+      delegate :connection, to: ActiveRecord::Base
+
+      def prev_meta
```

Comment:
Looks a bit complicated.

Can't we just store the previous meta as a thread-local variable and avoid DB queries and parsing?

Reply:
Using anything that involves threads doesn't sound simple to me at all 😅 Just my two cents

Maybe we can somehow use postgres `||` operator in `set_meta_param` method? It can merge two jsonb values. Although I had no luck applying it, maybe there is some postgres wizardry, that would allow us to do that.

Reply:
Threads are not evil)

Keeping the current value in the thread is the same as keeping it in the current postgres session in terms of potential problems) But thread is _local_